solutions/advanced/cses-1655.mdx

```diff
@@ -0,0 +1,89 @@
+---
+id: cses-1655
+source: CSES
+title: Maximum Xor Subarray
+author: Mihnea Brebenel
+---
+
+## Explanation
+
+Let $\texttt{pref}_i$ be the xor-sum of the prefix ending at $i$. Then, we must find a previous prefix sum
+$\texttt{pref}_j$ such that $\texttt{pref}_i \oplus \texttt{pref}_j$ is maximized. To obtain the maximal result
+the $0$ bits should be flipped to $1$. And this is where a trie data structure comes in handy. It can be used
+to maintain the binary representations of the xor-sums of prefixes.
```

Comment:
i think it would be useful to describe how we are gonna use a trie to solve the problem

like, maybe specify how we walk down the trie to get the best answer (and how we construct the trie to be from most significant to least signficant bits)

Reply:
ok i added a more in-depth explanation